docker-entrypoint.sh

```diff
@@ -64,6 +68,17 @@ else
 		AUTOINSTALL="false"
 	fi
 
+	echo "Config preprocessing before install ..."
+	sed -i \
```

Comment:
Do I understand this sed command correctly? Do are replacing `getenv(...)` parts with the actual values from environment-variables right? Whats the reason behind this? It seems quite complicated and error-prone to me.

Reply:
Yes, you are correct.

Basically, it's a refactoring in progress since env var are not visible from the supervisor commands.

Right now I'm trying to find a good way to allow a dynamic configuration that does not impact performances.
My current idea is to have the `redis`, `cache` and other dynamic config empty by default and to generate & replace the content in the sed.

This way, the common config would still be static (no performance impacts) and compatible with the "_simple_" setup (no Redis) while allowing advanced configuration.

Just need to find the time to implement and test this.

Reply:
Just an idea: We could use php to make layered config-files without much custom logic using `array_merge` and `var_export`. So we could provide a basic config, layering cache, redis, whatever configs on top on demand (based on env-vars).

Reply:
Not sure to get the idea. You mean to have like one file for each "_type_" of config and then merge them into the main `common.php` ?

If we did that, we would still need to need some kind of switch in the `common.php` to decide wether we need to merge a specific config.

Also, what's your use case for `var_export` exactly ?

Reply:
The humhub config files are just plain PHP Arrays so we could write a small php-cli script that is called during the startup which "merges" the right configs (based on env-vars), also reading env-vars with getenv() generating an combined php-array. This array can then be returned using `var_export`. The startup-script can then write the config to the actual config-file adding `return` and `<?php` `?>`-tags.

Reply:
Understood. Will try this.